ext/FluxCUDAExt/functor.jl

```diff
@@ -1,5 +1,27 @@
-
 adapt_storage(to::FluxCUDAAdaptor, x) = CUDA.cu(x)
+function adapt_storage(to::FluxCUDAAdaptor, x::AbstractArray)
+    typeof(to.ordinal) <: Nothing && return CUDA.cu(x)
+
+    # remember current device
+    old_ordinal = CUDA.device().handle
+
+    if !(x isa CuArray)
+        CUDA.device!(to.ordinal)
+        x_new = CUDA.cu(x)
+        CUDA.device!(old_ordinal)
+        return x_new
+    else
+        if CUDA.device(x).handle == to.ordinal
+            return x
+        else
+            CUDA.device!(to.ordinal)
+            x_new = CUDA.rand(size(x)...)
+            copy!(x_new, x)
+            CUDA.device!(old_ordinal)
+            return x_new
+        end
+    end
```

Comment:
```suggestion
    elseif CUDA.device(x).handle == to.ordinal
        return x
    else
        CUDA.device!(to.ordinal)
        x_new = copy(x)
        CUDA.device!(old_ordinal)
        return x_new
    end
```
I think this should work